### PR TITLE
add contour support

### LINKED ISSUE #605: not work with contour ingress
1. install contour
kubectl apply -f https://projectcontour.io/quickstart/contour.yaml
2. check external-ip
k8s get svc -n projectcontour envoy
3. install harbor
kubectl apply -f manifests/samples/harbor_full_fs.yaml
4. curl harbor failed
curl https://core.60.60.1.6.nip.io
`443: Connection refused`
5. curl http://core.60.60.1.6.nip.io
image

### PR DESCRIPTION
**Pull request by bitsf**
close #605

### CODE REVIEW

File: pkg/cluster/controllers/storage/ingress.go [hunk 1]
```diff
@@ -122,12 +127,16 @@ func (m *MinIOController) generateIngress(ctx context.Context, harborcluster *go
 	annotations["nginx.ingress.kubernetes.io/proxy-body-size"] = "0"
 
 	if harborcluster.Spec.Expose.Core.Ingress.Controller == v1alpha1.IngressControllerNCP {
-		annotations["ncp/use-regex"] = "true"
-		annotations["ncp/http-redirect"] = "true"
+		annotations["ncp/use-regex"] = NCPIngressValueTrue
+		annotations["ncp/http-redirect"] = NCPIngressValueTrue
+	} else if harborcluster.Spec.Expose.Core.Ingress.Controller == v1alpha1.IngressControllerContour {
+		annotations["ingress.kubernetes.io/force-ssl-redirect"] = ContourIngressValueTrue
```
Comment (heww) [inline pkg/cluster/controllers/storage/ingress.go:133]: `force-ssl-redirect` only when expose tls enabled.

File: pkg/cluster/controllers/storage/ingress.go [hunk 2]
```diff
@@ -122,12 +127,16 @@ func (m *MinIOController) generateIngress(ctx context.Context, harborcluster *go
 	annotations["nginx.ingress.kubernetes.io/proxy-body-size"] = "0"
 
 	if harborcluster.Spec.Expose.Core.Ingress.Controller == v1alpha1.IngressControllerNCP {
-		annotations["ncp/use-regex"] = "true"
-		annotations["ncp/http-redirect"] = "true"
+		annotations["ncp/use-regex"] = NCPIngressValueTrue
+		annotations["ncp/http-redirect"] = NCPIngressValueTrue
```
Comment (heww) [inline pkg/cluster/controllers/storage/ingress.go:131]: `ncp/http-redirect` only when expose tls enabled.

File: pkg/cluster/controllers/common/utils.go [hunk 3]
```diff
@@ -12,12 +12,12 @@ import (
 
 func GetIngressPath(ingress v1alpha1.IngressController) (string, error) {
 	switch ingress {
-	case v1alpha1.IngressControllerDefault:
+	case
```
Comment (steven-zou) [inline pkg/cluster/controllers/common/utils.go:15]: Do we still need this switch case? The kb spec annotation has indicated that the controller value will be limited in the enum and open API will validate them. So, the value here should be a known controller. Am I right?

There are some similar cases above.

**Review by steven-zou — COMMENTED** [review]
a comment

File: pkg/cluster/controllers/common/utils.go (hunk 3 above)
Comment (bitsf) [reply pkg/cluster/controllers/common/utils.go:15]: yes, it can be removed

**Review by heww — APPROVED** [review]

**Review by steven-zou — APPROVED** [review]
LGTM

**Pull request merged**